atoma-proxy/src/server/middleware.rs

```diff
@@ -1962,17 +2036,57 @@ pub mod auth {
                 .await;
         };
 
-        // NOTE: At this point, we have an acquired stack lock, so we can safely acquire a new stack.
-        acquire_new_stack(
-            state.state_manager_sender.clone(),
-            user_id,
-            lock_guard,
-            endpoint.to_string(),
-            total_tokens,
-            Arc::clone(&state.sui),
-            node,
-        )
-        .await
+        // We don't have a stack for the user, lets check if the user is using fiat currency.
+        let (result_sender, result_receiver) = oneshot::channel();
```

Comment:
Encapsulate this logic into a function, this possibly needs to be spawned on a new task as well ? For no cancellations hold state ?

Reply:
This is already inside tokio task called from authenticate_middleware.